helix/machine_learning/models/KANs.py

```diff
@@ -0,0 +1,907 @@
+            seed=seed,
+            epochs=epochs,
+            loss_fn=None,
+            # loss_fn=CrossEntropyLoss(),
+            lr=lr,
+            batch=batch,
+            problem_type=ProblemTypes.Classification,
+            loading_model=loading_model,
+        )
+        self.classes = None
+        self.classes_ = None
+        if loading_model:
+            self.classes_ = classes
+
+    def fit(self, X, y):
+
+        super().fit(X, y)
+        classes = np.unique(y).tolist()
+        self.classes_ = classes
+        self.classes = classes
+
+        return self
+
+    def predict(self, X):
+        y_pred_prob = self.predict_proba(X)
+        y_pred = np.argmax(y_pred_prob, axis=1)
+        return y_pred
+
+    def predict_proba(self, X):
+        X = self.sanitise_X(X)
+        y_pred_prob = self.forward(X).detach().numpy()
+        return y_pred_prob
+
+
+class KANRegressor(RegressorMixin, BaseEstimator, KANMixin):
+
+    def __init__(
+        self,
+        width: list = None,
+        grid: int = 5,
+        k: int = 3,
+        epochs: int = 100,
+        lr: float = 1,
+        batch: int = -1,
+        seed: int = 42,
+        loading_model=False,
+    ):
+
+        super().__init__(
+            width=width,
+            grid=grid,
+            k=k,
+            seed=seed,
+            epochs=epochs,
+            loss_fn=None,
+            lr=lr,
+            batch=batch,
+            problem_type=ProblemTypes.Regression,
+            loading_model=loading_model,
+        )
+
+    def predict(self, X):
+
+        X = self.sanitise_X(X)
+        y_pred = self.forward(X).detach().numpy().ravel()
+
+        return y_pred
+
+
+def f_inv(x, y_th):
+    x_th = 1 / y_th
+    return (
+        x_th,
+        y_th / x_th * x * (torch.abs(x) < x_th)
+        + torch.nan_to_num(1 / x) * (torch.abs(x) >= x_th),
+    )
+
+
+def f_inv2(x, y_th):
+    x_th = 1 / y_th ** (1 / 2)
+    return (
+        x_th,
+        y_th * (torch.abs(x) < x_th)
+        + torch.nan_to_num(1 / x**2) * (torch.abs(x) >= x_th),
+    )
+
+
+def f_inv3(x, y_th):
+    x_th = 1 / y_th ** (1 / 3)
+    return (
+        x_th,
+        y_th / x_th * x * (torch.abs(x) < x_th)
+        + torch.nan_to_num(1 / x**3) * (torch.abs(x) >= x_th),
+    )
+
+
+def f_inv4(x, y_th):
+    x_th = 1 / y_th ** (1 / 4)
+    return (
+        x_th,
+        y_th * (torch.abs(x) < x_th)
+        + torch.nan_to_num(1 / x**4) * (torch.abs(x) >= x_th),
+    )
+
+
+def f_inv5(x, y_th):
+    x_th = 1 / y_th ** (1 / 5)
+    return (
+        x_th,
+        y_th / x_th * x * (torch.abs(x) < x_th)
+        + torch.nan_to_num(1 / x**5) * (torch.abs(x) >= x_th),
+    )
+
+
+def f_sqrt(x, y_th):
+    x_th = 1 / y_th**2
+    return (
+        x_th,
+        x_th / y_th * x * (torch.abs(x) < x_th)
+        + torch.nan_to_num(torch.sqrt(torch.abs(x)) * torch.sign(x))
+        * (torch.abs(x) >= x_th),
+    )
+
+
+def f_power1d5(x, y_th):
+    return torch.abs(x) ** 1.5
+
+
+def f_invsqrt(x, y_th):
```

Comment:
```suggestion
def _f_invsqrt(x, y_th):
```
Leading underscore to indicate this is private to this module. Change it in the dict below too